sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/context/AuthTokenContext.java

```diff
@@ -118,8 +123,10 @@ public static synchronized void unsetInstance() {
     /**
      * Set or unset the refresh token update listener. There can be only 1 active.
      */
-    public void setRefreshListener(@NonNull RefreshListener refreshListener) {
-        if (!mRefreshListener.compareAndSet(null, refreshListener)) {
+    public synchronized void setRefreshListener(@NonNull RefreshListener refreshListener) {
+        if (mRefreshListener.compareAndSet(null, refreshListener)) {
+            mLastTokenRefreshed = null;
```

Comment:
I don't understand why we need to reset that when setting up the listener the first time. Plus it introduces synchronization on a path where use another concurrency mechanism instead.

Reply:
I'm less sure of it now than I was before. But the thought is to handle the scenario in `sameAuthTokenRefreshedAgainIfNoListenerOnFirstTry`

Reply:
Synced offline, this is not correct the changes in this function.